lms/djangoapps/instructor/views/api.py

```
@@ -1493,9 +1493,41 @@ def post(self, request, course_id, csv=False):  # pylint: disable=redefined-oute
             query_features = [
                 'id', 'username', 'name', 'email', 'language', 'location',
                 'year_of_birth', 'gender', 'level_of_education', 'mailing_address',
-                'goals', 'enrollment_mode', 'last_login', 'date_joined', 'external_user_key'
+                'goals', 'enrollment_mode', 'last_login', 'date_joined', 'external_user_key',
+                'enrollment_date',
             ]
 
+        custom_attributes = configuration_helpers.get_value_for_org(
+            course_key.org,
+            "student_profile_download_custom_student_attributes"
+        )
+        if custom_attributes:
+            # Validate type: must be list/tuple of strings.
+            if isinstance(custom_attributes, (list, tuple)):
+                if not all(isinstance(v, str) for v in custom_attributes):
+                    return JsonResponseBadRequest(
+                        _('Invalid custom student attribute configuration: all entries must be strings.')
+                    )
+                # Normalize (strip), drop empties, deduplicate preserving order.
+                cleaned = []
+                seen = set()
+                for v in custom_attributes:
+                    s = v.strip()
+                    if s and s not in seen:
+                        seen.add(s)
+                        cleaned.append(s)
```

This is a bit overkill and can also lead to confusion.  We shouldn't "magically" fix things for the end users.  If things are incorrect, we should inform the user that they are incorrect and fail fast.  Instead of correcting the list, we should just indicate if there are attributes that aren't valid `available_features` to the end user so they can correct their configuration.  Auto correcting makes the experience more limiting, what if they do in fact want the same column listed twice in their output because of something they do in their downstream processing, we don't want to remove that feature in a surprising way.

I think if any of the custom attributes is not in the available list, we can inform the user via an error message otherwise we should do what the user asked for.

---

Sounds good for me @feanil , I'll change it to let it fail instead of fixing it